go/libraries/doltcore/env/actions/table.go

```diff
@@ -189,12 +191,49 @@ func checkoutTablesAndDocs(ctx context.Context, dEnv *env.DoltEnv, roots map[Roo
 		}
 	}
 
-	err := dEnv.UpdateWorkingRoot(ctx, currRoot)
+	// update the working root with currRoot
+	_, err := env.UpdateWorkingRoot(ctx, dbData.Ddb, dbData.Rsw, currRoot)
+
+	return err
+}
+
+func checkoutDocs(ctx context.Context, dbData env.DbData, roots map[RootType]*doltdb.RootValue, docs doltdocs.Docs) error {
+	currRoot := roots[WorkingRoot]
+	staged := roots[StagedRoot]
+	head := roots[HeadRoot]
+
+	if len(docs) > 0 {
+		currRootWithDocs, stagedWithDocs, updatedDocs, err := getUpdatedWorkingAndStagedWithDocs(ctx, currRoot, staged, head, docs)
+		if err != nil {
+			return err
+		}
+		currRoot = currRootWithDocs
+		staged = stagedWithDocs
+		docs = updatedDocs
+	}
+
+	_, err := env.UpdateWorkingRoot(ctx, dbData.Ddb, dbData.Rsw, currRoot)
+	if err != nil {
+		return err
+	}
+
+	return dbData.Drw.WriteDocsToDisk(docs)
+}
+
+func checkoutTablesAndDocs(ctx context.Context, dbData env.DbData, roots map[RootType]*doltdb.RootValue, tbls []string, docs doltdocs.Docs) error {
```

Comment:
Ditto this.

Maybe checkout.go?

Reply:
Yeah i agree it's time for a checkout.go